hudi-common/src/main/java/org/apache/hudi/metadata/HoodieTableMetadataUtil.java

```diff
@@ -2389,7 +2395,12 @@ public static HoodieData<HoodieRecord> convertFilesToPartitionStatsRecords(Hoodi
       LOG.warn("No columns to index for partition stats index");
       return engineContext.emptyHoodieData();
     }
-    LOG.debug("Indexing following columns for partition stats index: {}", columnsToIndex);
+    // filter columns with only supported types
+    final List<String> validColumnsToIndex = columnsToIndex.stream()
```

Comment:
as part of this https://github.com/apache/hudi/pull/12310, we are enabling col stats for 3 meta fields (record key, commit time and partition path)

Reply:
So, we should include them as part of partition stats as well

Reply:
done